alerting/provider/slack/slack.go

```diff
@@ -46,11 +46,15 @@ func (provider *AlertProvider) Send(endpoint *core.Endpoint, alert *alert.Alert,
 // buildRequestBody builds the request body for the provider
 func (provider *AlertProvider) buildRequestBody(endpoint *core.Endpoint, alert *alert.Alert, result *core.Result, resolved bool) string {
 	var message, color, results string
+	name := endpoint.Name
+	if endpoint.Group != "" {
+		name = endpoint.Group + ":" + name
+	}
 	if resolved {
-		message = fmt.Sprintf("An alert for *%s* has been resolved after passing successfully %d time(s) in a row", endpoint.Name, alert.SuccessThreshold)
+		message = fmt.Sprintf("An alert for *%s* has been resolved after passing successfully %d time(s) in a row", name, alert.SuccessThreshold)
 		color = "#36A64F"
 	} else {
-		message = fmt.Sprintf("An alert for *%s* has been triggered due to having failed %d time(s) in a row", endpoint.Name, alert.FailureThreshold)
+		message = fmt.Sprintf("An alert for *%s* has been triggered due to having failed %d time(s) in a row", name, alert.FailureThreshold)
```

Comment:
Personally, I'd prefer having something like 
```
message = fmt.Sprintf("An alert for *%s* in group *%s* has been resolved after passing successfully %d time(s) in a row", endpoint.Name, endpoint.Group, alert.SuccessThreshold)
```

I started working on adding this in the other providers on my local branch :P

Reply:
On second thoughts, I'm a little worried about what group names people are using.
If any of these group names are using `\` or `"`, it might cause some parsing issues.

I have to head out, but I'll give it some thought.

Reply:
Did some thinking and I decided to implement it in every alerting providers, as well as add validation to prevent the usage of `"` and `\` in endpoint names, groups and alert descriptions.

Sorry for stealing your PR! 😅